doc/contribute.rst

```diff
@@ -0,0 +1,252 @@
+.. _contribute:
+
+########################
+Develop and contribute
+########################
+
+Welcome to the RAMP team. We are always happy to have new RAMP developers.
+
+You can contribute to this code by making a `Pull Request 
+<https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests>`_ 
+on Github_. Please, make sure that your code is coming with unit tests to 
+ensure full coverage and continuous integration in the API.
+
+
+.. _GitHub: https://github.com/paris-saclay-cds/ramp-board/pulls
+
+
+Install for development
+-----------------------
+To install RAMP please fallow :ref:`install guideline <install>` making sure 
+that you use the ``make inplace`` option.
+
+
+Prepare database engine
+-----------------------
+Next, create the database, following the steps as described here:
+:ref:`create database <set_database>`
+
+You created the ``Postgres database cluster`` using the command::
+
+    ~ $ initdb postgres_dbs
+
+and then you started it with::
+
+    ~ $ pg_ctl -D postgres_dbs -l logfile start
+
+However, RAMP tests do not know which database engine you are using for 
+your ramp development and therefore you need to inform RAMP tests about this
+To do this, open the ``db_engine.yml`` file located in the ``ramp-board`` directory. 
+It should look as follows::
+
+    db_owner: postgres
+    db_cluster_name: postgres
+
+The above settings are used by ``Travis Cl`` which is a continuous integration
+service hosted at ``GitHub``. Therefore, you need to change them to your local
+settings.
+
+``db_owner`` is the owner of your database. If you don't know who is 
+the owner of your database you can find it out by typing in your terminal::
+    
+    ~ $ psql -l
+
+This command will list all of your databases along with their owners. 
+
+``db_cluster_name`` is the name of your Postgres database cluster (if you used
+commands as above your ``Postgres database cluster`` name is 'postgres_dbs').
+
+Test
+----
+Each time before running the tests make sure your Postgres database cluster has been
+started. You can always start it using the command::
+
+    ~ $ pg_ctl -D postgres_dbs -l logfile start
+
+You will also need SMTP server, to run it type::
```

Comment:
```suggestion
In addition, you might want to start an SMTP server to run all the tests.
If you don't run the server, some of the tests will be marked as `xfail` 
because they cannot be run. To launch the server, execute the following 
command::`
You will also need SMTP server, to run it type::
```

Reply:
atm the tests fail if there is no smtp server because there is no way of checking if smtp server is on at the test side